### ISSUE TITLE
Problem with python-indent-offset

### ISSUE THREAD

**Issue by Pitometsu** [mattfidler/auto-indent-mode.el#53]
When I have
```elisp
'(auto-indent-assign-indent-level 2)
```
and
```elisp
'(python-indent-offset 4)
```
And do `uto-indent-mode-on`, `auto-indent-mode` break `python-indent-offest` by setting it to `2`, and in **\*Customize Group: Python\*** it marked as `CHANGED outside Customize.`.

***

What expected is: re-use real `python-indent-offset` value.

`GNU Emacs 25.3.1 (x86_64-apple-darwin17.2.0, NS appkit-1561.10 Version 10.13.1 (Build 17B1003)) of 2017-12-11`

**Comment by Pitometsu** [comment]
@mattfidler is it plugin maintained yet?

**Comment by mattfidler** [comment]
I really don't maintain as I ought to currently....

However I still use emacs and auto indent mode.  I looked at other modes to, but I didn't like them... 

There may be a way to fix this.

**Comment by Pitometsu** [comment]
I gues, in `Auto Indent Known Indent Level Variables` `python-indent` should be replaced by `pytho-indent-offset`.

**Comment by Pitometsu** [comment]
@mattfidler, yeah, I really like this mode too, so want to believe it will live ;-)

**Comment by Pitometsu** [comment]
…And set `auto-indent-assign-indent-level-variables` to `nil`

**Comment by Pitometsu** [comment]
…But  now it does not do actually auto-indentation for python %)

**Comment by Pitometsu** [comment]
`reindent-then-newline-and-indent` works, but not called

**Comment by mattfidler** [comment]
Python auto indent is a bit hard. I will look into adding these new variables.  I think there was an option to turn it on...  It was more than a bit experimental.

You could submit a pull request if you get to it sooner

**Comment by Pitometsu** [comment]
Also it may be reasonable to disable electric mode when AC enabled
`(setq electric-indent-inhibit t)`.

**Comment by mattfidler** [comment]
I added the things that you requested.

**Comment by mattfidler** [comment]
See:

https://github.com/mattfidler/auto-indent-mode.el/issues/27

This is the reason things were set globally...

**Comment by mattfidler** [comment]
I'm not sure if `auto-indent` should match the indentation in python or not.

**Comment by mattfidler** [comment]
Also for me, the `python-mode` is working.  I have not yet upgraded emacs... I have 25.2; I have heard that 26 is out...?

**Comment by Pitometsu** [comment]
Thank you, I'll check. I'm on 25.3 yet.
